src/region.js

```diff
@@ -204,12 +208,38 @@ Marionette.Region = Marionette.Object.extend({
     return Backbone.$(el, Marionette._getValue(this.options.parentEl, this));
   },
 
+  // Replace the region's DOM element with the view's DOM element.
+  replaceEl: function(view) {
+    // empty el so we don't save any non-destroyed views
+    this.$el.contents().detach();
```

Comment:
Ok so this was added to handle the case where replaceEl was used with preventDestroy.

If we don't remove the old view in here when we restore it the old view will come back as we haven't destroyed it.

I just took this from attachHtml - I'll try with remove